### PR TITLE
Add branch name-copy buttons on the branches list

### PR DESCRIPTION
**Pull request by puni9869**
Fix:: #25120

There are some observation in different platforms like Github, Gitlab, Bitbucket  and bunch of more platform. 
I have observed they all have copy option for branch name. They don't have copy option for tags.

As a UX prespective for tags we don't need copy option. 

By above observation I am suggsting the following UI changes, I know they might not align with UX point of way.But they are simplest to make quick.

Here is my suggestions:
image

When Hover on Copy
image

When Clicked on Copy
image

Code changes in files:- 

```html
<button class="btn interact-bg gt-p-3" data-clipboard-text="{{.Name}}" data-tooltip-content="{{$.locale.Tr "repo.branch.copy_branch_name"}}">{{svg "octicon-copy"}}</button>
```
Translations:- 
en-US.ini
```ini
[repo]
branch.copy_branch_name = Copy Branch Name
```

@silverwind  @delvh 
I am open for any UI/UX suggestion which work the best.

### CODE REVIEW

File: templates/repo/branch/list.tmpl [hunk 1]
```diff
@@ -131,6 +131,7 @@
 										{{end}}
 									</td>
 									<td class="three wide right aligned overflow-visible">
+									<button class="btn interact-bg gt-p-3" data-clipboard-text="{{.Name}}" data-tooltip-content="{{$.locale.Tr "repo.branch.copy_branch_name"}}">{{svg "octicon-copy"}}</button>
```
Comment (silverwind) [inline templates/repo/branch/list.tmpl]:
```suggestion
										<button class="btn interact-bg gt-p-3" data-clipboard-text="{{.Name}}" data-tooltip-content="{{$.locale.Tr "repo.branch.copy_branch_name"}}">{{svg "octicon-copy"}}</button>
```

**Review by silverwind — DISMISSED** [review]

**Review by delvh — APPROVED** [review]

**Comment by techknowlogick** [comment]
@puni9869 could you check the "allow edits from maintainers" checkbox on this PR?

**Comment by lunny** [comment]
How about moving the copy button following the branch name? There are too many buttons at the end of this line.

**Comment by silverwind** [comment]
> How about moving the copy button following the branch name? There are too many buttons at the end of this line.

I think that would look strange. Imho it looks better when the buttons are together.

File: options/locale/locale_en-US.ini [hunk 2]
```diff
@@ -2474,6 +2474,7 @@ branch.create_branch_operation = Create branch
 branch.new_branch = Create new branch
 branch.new_branch_from = Create new branch from "%s"
 branch.renamed = Branch %s was renamed to %s.
+branch.copy_branch_name = Copy Branch Name
```
Comment (wxiaoguang) [inline options/locale/locale_en-US.ini:2478]: The text could be `Copy branch name`, to match other texts like `Create new branch`

Comment (silverwind) [inline options/locale/locale_en-US.ini:2478]:
```suggestion
branch.copy_branch_name = Copy branch name
```

Comment (silverwind) [reply options/locale/locale_en-US.ini:2478]: See suggestion below

**Comment by lunny** [comment]
> > How about moving the copy button following the branch name? There are too many buttons at the end of this line.
> 
> I think that would look strange. Imho it looks better when the buttons are together.

I have the opposite opinion. I don't think putting too many buttons together is a good design. And look like all other places, copy button is near the content.

**Comment by silverwind** [comment]
We can move it after branch name, with a styling like in diff header, e.g. no padding:

Screenshot 2023-06-29 at 06 28 47

**Comment by puni9869** [comment]
Due to conflicts. Pr is moved to https://github.com/go-gitea/gitea/pull/25596

**Pull request closed without merging**